src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/AbstractRemoveEmptyFinalizers.cs

```diff
@@ -35,72 +33,26 @@ public override void Initialize(AnalysisContext analysisContext)
             analysisContext.EnableConcurrentExecution();
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            analysisContext.RegisterCompilationStartAction(compilationContext =>
+            analysisContext.RegisterSymbolAction(methodContext =>
```

Comment:
Sure, but code block could potentially get called back multiple times for a valid finalizer, right?
```csharp
class C1
{
    ~C1() 
    {
        if (true) 
        {
            DisposeOfSomething();
        }
    }
}
```